Documentation/ProjectArchitecture.md

```diff
@@ -0,0 +1,41 @@
+# Project architecture
+
+Our single app projects use the same basic module architecture and consist from the following base targets. 
+
+Basically any of the following modules can contain tests and "testing" code. For us "testing" code means any useful extensions and mock implementations that could be used for tests. To simplify usage of this testing code we have optional "Testing" folder under module's folder (that usually contains Sources and Tests folder) which is imported for debug configurations. This way we make sure this testing code never leaves dev machines.
+
+## Core
+Contains stuff that is shared basically throughout the whole project - models, extensions. Might be renamed based on your project name - _\<Project>Core_, _\<Project>\_Core_, depends on preferences of the dev team.
+
+Most importantly it contains protocol interfaces **without any implementations**. In the [example project][example project] you can see [`UserManager`](/ProjectTemplate/Modules/Core/Sources/Protocols/UserManager.swift), that defines how user manager looks like and that's all of it. If we were really correct, we should rather have a module for that but sometimes it seems like too much effort that would significantly increase number of dynamic frameworks so this is our little compromise as basically all modules depend on Core. Of course you can do it your way 😎
+
+## AppUI
+Contains UI components that are shared throughout the app - button styles, text styles, colors, reused list cells, etc. By design this module does not depend on [Core](#core) as we prefer these view components to be just stupid UI views.
```

Comment:
nitpick: maybe we can use `simple` instead of `stupid`, which may sound a little bit harsh? 😄